src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -2742,18 +2741,19 @@ class ArnoldGlobals
 			}
 			AiNodeSetPtr( options, g_cameraArnoldString, arnoldCamera );
 
-			const IECore::V2iData *resolution = cortexCamera->parametersData()->member<IECore::V2iData>( "resolution" );
-			AiNodeSetInt( options, g_xresArnoldString, resolution->readable().x );
-			AiNodeSetInt( options, g_yresArnoldString, resolution->readable().y );
+			Imath::V2i resolution;
+			bool hasRegion;
+			Imath::Box2i renderRegion;
+			cortexCamera->renderImageSpec( resolution, hasRegion, renderRegion );
 
-			const IECore::FloatData *pixelAspectRatio = cortexCamera->parametersData()->member<IECore::FloatData>( "pixelAspectRatio" );
-			AiNodeSetFlt( options, g_pixelAspectRatioArnoldString, pixelAspectRatio->readable() );
+			AiNodeSetInt( options, g_xresArnoldString, resolution.x );
+			AiNodeSetInt( options, g_yresArnoldString, resolution.y );
 
-			const IECore::Box2iData *renderRegion = cortexCamera->parametersData()->member<IECore::Box2iData>( "renderRegion" );
+			AiNodeSetFlt( options, g_pixelAspectRatioArnoldString, cortexCamera->getPixelAspect() );
 
-			if( renderRegion )
+			if( hasRegion )
```

Comment:
Referring back to our previous conversation about the utility of `hasRegion`, it seems that we could perhaps simplify this code if we didn't use `hasRegion`, and just always unconditionally set the render region (assuming we were given the region of the full resolution when there was no crop/overscan).

Reply:
Talking to Don about this, he does slightly prefer the current logic where if you don't specify any render options to create a render region, we don't pass a render region to Arnold.  His recommendation was that if the only object was API I should create a Camera::RenderImageSpec struct with the 3 values and then bind that to Python.  This sounds like enough work that if you would prefer that I be lazy and always bake in the render region, I guess I won't object.

Reply:
My preference for the Camera API is still this :

```
V2i renderResolution() const;
Box2i renderRegion() const; // Always returns the region, even if crop/overscan is off
```

It avoids awkward output parameters and a struct that bundles things together unnecessarily. If you just want the resolution, you can get it without passing two extra parameters you don't care about. And if you want the render region, you can get it without a special check to see if it is valid and then fall back to a `Box2i( 0, 0, width, height )` you make manually yourself. And if you want to know if the render region is different than the resolution, you can compare them. It's a little bit more work to say "is the render region doing something special?", but I don't see that as important enough to dictate an otherwise awkward API. In the end, we never make ass files, let alone look at them, so going out of our way to unset the region rather than always set it to width/height doesn't really buy us anything...